### PR TITLE
MK-1713 - manage RQLNode.FILTER containing a RQLNode

### PR DESCRIPTION
**Pull request by meek0**

### CODE REVIEW

File: src/main/java/org/obiba/es/mica/query/RQLQuery.java [hunk 1]
```diff
@@ -185,11 +185,11 @@ private void parseNode(ASTNode node) {
               case FIELDS:
                 parseFields(n);
                 break;
-              case MATCH:
-                if (n.getArgumentsSize() == 1) {
-                  parseFullTextMatch(n);
-                } else {
-                  parseQuery(n);
+              case FILTER: // for now will only have one argument RQL_NODE.MATCH
```
Comment (ymarcon) [inline src/main/java/org/obiba/es/mica/query/RQLQuery.java]: Why is there such restriction?

Comment (ymarcon) [reply src/main/java/org/obiba/es/mica/query/RQLQuery.java]: It's not a "fullTextMatchQuery" that you are building, but a generic filterQuery. Whether it is a full-text match query is a client concern.

**Review by ymarcon — CHANGES_REQUESTED** [review]

**Comment by meek0** [comment]
Kept the one argument restriction but removed the RQLNode.MATCH restriction

File: src/main/java/org/obiba/es/mica/query/RQLQuery.java [hunk 2]
```diff
@@ -185,19 +184,18 @@ private void parseNode(ASTNode node) {
               case FIELDS:
                 parseFields(n);
                 break;
-              case MATCH:
-                if (n.getArgumentsSize() == 1) {
-                  parseFullTextMatch(n);
-                } else {
-                  parseQuery(n);
+              case FILTER: // for now will only have one argument
```
Comment (ymarcon) [inline src/main/java/org/obiba/es/mica/query/RQLQuery.java]: Why?

**Review by ymarcon — APPROVED** [review]

**Pull request merged**